frontend/src/components/panels/WordCloudPanel.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Box, Paper, Typography } from '@mui/material';
+import { IWordFrequency } from 'shared-types';
+import { Text } from '@visx/text';
+import { scaleLog } from '@visx/scale';
+import Wordcloud from '@visx/wordcloud/lib/Wordcloud';
+import WordCloudSkeleton from '../skeletons/WordCountSkeleton';
+
+interface WordCloudPanelProps {
+  data: IWordFrequency[];
+  height: number;
+  loading: boolean;
+  colors: string[];
+}
+
+const WordCloudPanel: React.FC<WordCloudPanelProps> = ({ data, height, loading, colors }: WordCloudPanelProps) => {
+  const [containerWidth, setContainerWidth] = useState<number>(0);
+
+  const handleBoxWidth = useCallback((node?: HTMLDivElement) => {
+    if (node) {
+      setContainerWidth(node.getBoundingClientRect().width);
+    }
+  }, []);
+
+  const fontScale = useMemo(
+    () =>
+      scaleLog({
+        domain: [Math.min(...data.map((w) => w.value)), Math.max(...data.map((w) => w.value))],
```

Comment:
just mice to know that lodash has `maxBy` and `minBy` functions which are very nice in this case